server/src/test/java/org/elasticsearch/action/search/TransportSearchActionTests.java

```diff
@@ -1059,6 +1060,21 @@ public void testShouldMinimizeRoundtrips() throws Exception {
             searchRequest.setCcsMinimizeRoundtrips(false);
             assertFalse(TransportSearchAction.shouldMinimizeRoundtrips(searchRequest));
         }
+        {
+            SearchRequest searchRequest = new SearchRequest();
+            SearchSourceBuilder source = new SearchSourceBuilder();
+            source.knnSearch(new KnnSearchBuilder("field", new float[] { 1, 2, 3 }, 10, 50));
+            searchRequest.source(source);
+
+            // Emulate TransportSearchAction logic: first adjust search type, then check minimize roundtrips
+            TransportSearchAction.adjustSearchType(searchRequest);
+
+            // Minimize roundtrips should always be false because kNN uses DFS_QUERY_THEN_FETCH
```

Comment:
too bad that we can't minimize roundtrips! maybe that's a good reminder to rethink whether we can do something about this, I will try to think about this.

Reply:
In a follow-up, I hope to optimize the case where there is only kNN, with no query or aggregations. Then we can run kNN during the main query phase (using QUERY_THEN_FETCH), and it will be possible to minimize roundtrips.

Reply:
I see, agreed that it's a bit hacky because it requires adjustSearchType to be called. I would rather add a very simple unit test around adjustSearchType then. this specific test method already tests the scenario with dfs enabled, so we should just verify that the search type is adjusted when knn is used.

Reply:
Also, Nhat is running benchmarks to asses how bad it is that we don't minimize roundtrips in some cases, in the context of https://github.com/elastic/elasticsearch/issues/73971 . I would imagine that the results of the benchmark will also inform whether we need to put more effort into minimizing roundtrips when knn is used.

Reply:
👍 I'll rework this test. Looking forward to seeing Nhat's results.